src/core/auth/components/LoginForm.tsx

```diff
@@ -1,32 +1,68 @@
+  handleChangePassword: ({ target }: ChangeEvent<HTMLInputElement>) => void;
+  handleSubmit: () => void;
+  isSubmitDisabled: boolean;
+  isLoading: boolean;
+  errorMessage: string;
+}
 
-  const handleChangeUserName = ({ target }: ChangeEvent<HTMLInputElement>) => {
-    dispatch(changeUserName(target.value));
-  };
+export const LoginForm = ({ handleChangeUserName, handleChangePassword, handleSubmit, isSubmitDisabled, isLoading, errorMessage }: LoginFormProps): JSX.Element => {
+  const [showPassword, setShowPassword] = useState(false);
 
-  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
-  //@ts-ignore
-  const from = location.state?.from?.pathname || routes.protected.dashboard;
+  const handleClickShowPassword = () => setShowPassword((show) => !show);
 
-  const submitHandle = () => {
-    dispatch(setLoggedIn());
-    navigate(from, { replace: true });
+  const handleMouseDownPassword = (event: React.MouseEvent<HTMLButtonElement>) => {
+    event.preventDefault();
   };
 
   return (
-    <div className="login-form">
-      <h1>Login 3</h1>
-      <input type="text" name="username" onChange={handleChangeUserName} />
-      <button onClick={submitHandle}>login</button>
-    </div>
+    <form className="login-form">
+      <div className="login-form-container login-form-input">
+        <div className="login-form__field-container login-form__field-container--lang">
+          <div className="language-setting__select"></div>
+        </div>
+        <div className="login-form__field-container login-form__field-container--username">
+          <TextField error={!!errorMessage} disabled={isLoading} fullWidth id="outlined-basic" label="Username" variant="outlined" type="text" name="username" onChange={handleChangeUserName} />
+        </div>
+        <div className="login-form__field-container login-form__field-container--password">
+          <FormControl disabled={isLoading} error={!!errorMessage} fullWidth id="outlined-basic" variant="outlined" onChange={handleChangePassword}>
+            <InputLabel disabled={isLoading} error={!!errorMessage} htmlFor="outlined-adornment-password">
+              Password
+            </InputLabel>
+            <OutlinedInput
+              error={!!errorMessage}
+              id="outlined-adornment-password"
+              type={showPassword ? 'text' : 'password'}
+              endAdornment={
+                <InputAdornment position="end">
+                  <IconButton disabled={isLoading} aria-label="toggle password visibility" onClick={handleClickShowPassword} onMouseDown={handleMouseDownPassword} edge="end">
+                    {showPassword ? <VisibilityOff /> : <Visibility />}
+                  </IconButton>
+                </InputAdornment>
+              }
+              label="Password"
+            />
+          </FormControl>
+        </div>
+        <div className="login-form-alert-box">{errorMessage && <Alert severity="error">{errorMessage}</Alert>}</div>
```

Comment:
Is there any reason to keep empty div.login-form-alert-box  in case when no errors?

Reply:
How else can you reserve the space for emerging alert and make sure your components stay in place when alert state is false?

Reply:
For example, the form layout could be positioned with the grid display mode.  Using the template-grid-row property, the inputs block and the button block could be justified properly within the whole available space. The alert block could be set to the relative position, meanwhile the alert block -to the  absolute one. Thus the alert should be excluded from the regular layout render, and be independent on the surrounding blocks.

Reply:
Styles reworked with implementation of grid. Alert block positioning has been set to absolute. However, as a side note for future development, the desktop version of the login form needs to have a revision and approval/ disapproval in terms of design and layout of input fields block - alert block - login button.